# HG changeset patch
# User Andreas Weber <andy.weber.aw@googlemail.com>
# Date 1371045830 14400
#      Wed Jun 12 10:03:50 2013 -0400
# Node ID 022dba56de1e78b660f036efda6fdfdd3cc34234
# Parent  f8f35ddc49a2648da55670ec9e5dccd53fab4220
eliminate unnecessary normalisation of view vector in surfl (bug #39241)

* surfl.m: Don't normalize view vector.

diff --git a/scripts/plot/surfl.m b/scripts/plot/surfl.m
--- a/scripts/plot/surfl.m
+++ b/scripts/plot/surfl.m
@@ -126,18 +126,17 @@ function retval = surfl (varargin)
     if (! ishold ())
       set (h, "view", [-37.5, 30],
            "xgrid", "on", "ygrid", "on", "zgrid", "on", "clim", [0 1]);
     endif
 
     ## Get view vector (vv).
     a = axis;
     [az, el] = view;
-    [vv(1), vv(2), vv(3)] = sph2cart ((az - 90) * pi/180.0, el * pi/180.0, 1.0);
-    vv /= norm (vv);
+    vv = sph2cart ((az - 90) * pi/180.0, el * pi/180.0, 1.0);
 
     if (!have_lv)
       ## Calculate light vector (lv) from view vector.
       Phi = 45.0 / 180.0 * pi;
       R = [cos(Phi), -sin(Phi), 0;
            sin(Phi),  cos(Phi), 0;
            0,          0,         1];
       lv = (R * vv.').';
